ui/src/app/base/components/SubnetField/SubnetField.tsx

```diff
@@ -0,0 +1,59 @@
+import { useEffect } from "react";
+
+import { Select, Spinner } from "@canonical/react-components";
+import { useDispatch, useSelector } from "react-redux";
+
+import FormikField from "app/base/components/FormikField";
+import type { Props as FormikFieldProps } from "app/base/components/FormikField/FormikField";
+import { actions as subnetActions } from "app/store/subnet";
+import subnetSelectors from "app/store/subnet/selectors";
+import type { Subnet } from "app/store/subnet/types";
+import { getSubnetDisplay } from "app/store/subnet/utils";
+
+type Props = {
+  defaultOption?: { label: string; value: string };
+  filterFunction?: (subnet: Subnet) => boolean;
+} & Partial<FormikFieldProps>;
+
+export const SubnetField = ({
+  defaultOption,
+  filterFunction,
+  ...props
+}: Props): JSX.Element => {
+  const dispatch = useDispatch();
+  let subnets: Subnet[] = useSelector(subnetSelectors.all);
```

Comment:
Does this need to be typed?

Reply:
For some reason these are coming back as `unknown` so when the `.map` happens below it says `Property 'map' does not exist on type 'unknown'`.

Reply:
Right, I guess it could be because it's declared as a `let`, which could change to be anything.

Reply:
Even with `let` you'd need to explicitly cast it to another type if you wanted to redeclare it with something else. I think it's more likely our selector util doesn't maintain the types for some reason.